worker/src/main/java/gov/cms/ab2d/worker/processor/JobDataWriterImpl.java

```diff
@@ -0,0 +1,208 @@
+package gov.cms.ab2d.worker.processor;
+
+import gov.cms.ab2d.common.model.Contract;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import static java.nio.file.StandardOpenOption.APPEND;
+
+/**
+ * Since the Worker processes contracts sequentially, create a new instance for each contract,
+ * and make sure only that single instance is shared among the threads.
+ */
+@Slf4j
+public class JobDataWriterImpl implements JobDataWriter {
+    private static final String OUTPUT_FILE_SUFFIX = ".ndjson";
+    private static final String ERROR_FILE_SUFFIX = "_error.ndjson";
+
+    private final Path outputDir;
+    private final Contract contract;
+    private final long maxFileSize;
+    private final int tryLockTimeout;
+
+    private final Lock lock = new ReentrantLock();
+
+    private Path errorFile;
+    private Path dataFile;
```

Comment:
We probably want `dataFile` to be `volatile` as well to be safe.

Reply:
done